init-pants/README.md

```diff
@@ -0,0 +1,80 @@
+# init-pants-action
+
+GitHub Action that installs pants and prepares the pants caches.
+
+Currently, this action only supports storing the pants caches in the GHA cache.
+Eventually, this should become configurable to support projects that use remote caching.
+
+This action manages three GHA caches:
+1. The pants `setup` cache.
+2. The pants `named_caches` cache.
+3. The pants `lmdb_store` cache.
```

Comment:
I'm somewhat skeptical of the value of caching lmdb_store, as it'll very quickly blow the 10GB cache limit. Basically every commit will have its own slightly different value for this, and that value will contain the entire lmdb_store at that point in time, most of which is irrelevant.

Reply:
I know we do this in the example-repo, but I'm skeptical of that too (and note that it hashes every file in the repo to generate the key), and putting that here elevates it to the level of best practice, which I think it is not... :)

Reply:
Maybe the tradeoffs should be explained in the README, and this turned off by default, but there is a flag for turning it on if the user thinks it's worth it?

Reply:
I saw how the example-python repo hashes all files, which sounds expensive. A git commit sha is effectively a hash of the entire repo, so I figured that was a better option.

To abandon a cache, we just update the `gha-cache-key`, and that will force the workflow to start with an empty cache.

I figured that as caches went unused, they'd just end up deleted. But I neglected to account for the 10GB cache max. For StackStorm, the rate of merging PRs is slow, so the 7 day limit on unused caches is probably sufficient. But a project that moves faster would run into problems.

I like the idea of disabling the lmdb_store cache by default and documenting this discussion about it, but still allowing it to be cached.

Won't that cover the remote caching use case as well? Just leave the lmdb_store cache disabled if you are using remote caching? Or would anything else be needed in this action to deal with remote caching?

Reply:
With remote caching there would be no need to do anything special to conserve lmdb_store, you could ignore the issue entirely.